src/Message/TransactionStatusRequest.php

```diff
@@ -42,4 +57,106 @@ public function sendData($data): ResponseInterface
             $this->getResponse()->getStatusCode()
         );
     }
+
+    /**
+     * Use the data sent by Icepay in the post back to check the status.
+     * This is necessary because Icepay has a delay in their backend if you request the status immediately after the signal.
+     *
+     * @see http://docs2.icepay.com/payment-process/handling-the-postback/postback-sample/
+     *
+     * @return TransactionStatusResponse|bool false when the data is is not sent or not correct
+     *
+     * @throws PostBackException by call to self::validateSecurityHashMatch()
+     */
+    private function getTransactionStatusFromPostBack()
+    {
+        if (stripos($this->httpRequest->getContentType(), 'json') === false) {
+            return false;
+        }
+
+        try {
+            $contentAsArray = json_decode($this->httpRequest->getContent(), true);
+        } catch (\LogicException $exception) {
+            return false;
+        }
+
+        if (is_array($contentAsArray) === false || isset($contentAsArray['StatusCode']) === false) {
+            return false;
+        }
+
+        $this->validateSecurityHashMatch($this->httpRequest, $contentAsArray);
+
+        $camelCasedKeysContent = array_combine(
+            array_map('lcfirst', array_keys($contentAsArray)),
+            array_values($contentAsArray)
+        );
+
+        return new TransactionStatusResponse(
+            $this,
+            $camelCasedKeysContent,
+            200
```

Comment:
Isn't it better to return the httpRequest status code instead of hardcoded 200?

Reply:
At this point no request is being sent to Icepay, so no response status is available.